executor/include/executor/best_fit.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception, Inc.
+ *  Author(s): Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#include <executor/type_trait.h>
+
+#include <executor/default/cuda_executor.hpp>
+#include <executor/default/inline_executor.hpp>
+#include <executor/default/omp_executor.hpp>
+#include <executor/default/sse_executor.hpp>
+
+namespace executor {
+
+static const auto best_fit_executors =
+    std::make_tuple(executor::omp_executor<>{}, executor::sse_executor<>{},
+                    executor::inline_executor<>{});
+
+struct executor_runtime_checks {
+  template <typename Executor, typename executor::InstanceOf<
+                                   Executor, executor::inline_executor> = 0>
```

Comment:
Can't you put multiple types on the same check?

Also, this should be in detail since this is not what the user is supposed to use but is a default 'value'

Reply:
You can, but wouldn't we want different checks per executor type? 
How would have multiple types on a single check be used?

I haven't put it in details, since if a user wants to specify his own checks he can do so by inheriting from `executor_runtime_checks`.

Reply:
The function isn't virtual so the downstream user's concerns can be ignored

Reply:
I would want this to be virtual ideally, but sadly templates and virtual don't play well.
I am not sure I understood why you want to combine them

Reply:
Them being combined doesn't affect the downstream user in any adverse manner
Combining reduces code
The code is really the same, so combining makes sense